src/content/app/entity-viewer/state/bookmarks/entityViewerBookmarksSlice.ts

```diff
@@ -70,6 +148,18 @@ const bookmarksSlice = createSlice({
           label: gene.symbol ? [gene.symbol, gene.stable_id] : gene.stable_id,
           type: 'gene' as const
         };
+
+        // clear the stored data for the oldest previously viewed entity
+        if (previouslyViewedEntities.length === 20) {
+          const oldestPreviouslyViewedEntity =
+            previouslyViewedEntities[previouslyViewedEntities.length - 1];
+
+          entityViewerStorageService.clearGeneViewTranscriptsState({
+            genomeId,
+            entityId: oldestPreviouslyViewedEntity.entity_id
+          });
+        }
```

Comment:
whoa! we should move this side-effecty code out of the reducer; probably into the updatePreviouslyViewedEntities thunk